cumulus/polkadot-parachain/src/service.rs

```diff
@@ -401,17 +410,23 @@ pub fn build_aura_import_queue(
 		_,
 	>(cumulus_client_consensus_aura::ImportQueueParams {
 		block_import,
-		client,
-		create_inherent_data_providers: move |_, _| async move {
-			let timestamp = sp_timestamp::InherentDataProvider::from_system_time();
+		client: client.clone(),
+		create_inherent_data_providers: move |_, _| {
+			let client = client.clone();
 
-			let slot =
+			async move {
+				let timestamp = sp_timestamp::InherentDataProvider::from_system_time();
+
+				let slot_duration = cumulus_client_consensus_aura::slot_duration(&*client)?;
```

Comment:
this change seems unrelated, what are the implications? added support for dynamic slot durations?

Reply:
Yes